core/main/src/broker/thunder_broker.rs

```diff
@@ -105,14 +156,43 @@ impl ThunderBroker {
                         }
 
                     },
-                    Some(request) = tr.recv() => {
+                    Some(mut request) = tr.recv() => {
                         debug!("Got request from receiver for broker {:?}", request);
-                        match broker_c.prepare_request(&request) {
-                            Ok(updated_request) => {
-                                debug!("Sending request to broker {:?}", updated_request);
-                                for r in updated_request {
-                                    let _feed = ws_tx.feed(tokio_tungstenite::tungstenite::Message::Text(r)).await;
-                                    let _flush = ws_tx.flush().await;
+
+                        match broker_c.check_and_generate_plugin_activation_request(&request) {
+                            Ok(requests) => {
+                                if !requests.is_empty() {
+                                    let mut ws_tx = ws_tx_wrap.lock().await;
+                                    for r in requests {
+                                        let _feed = ws_tx.feed(tokio_tungstenite::tungstenite::Message::Text(r)).await;
+                                        let _flush = ws_tx.flush().await;
+                                    }
+                                }
+                                else {
+                                    // empty request means plugin is activated and ready to process the request
+                                    // Intercept the request for data migration
+                                    let mut request_consumed = false;
+                                    if let Some(user_data_migrator) = broker_c.data_migrator.clone() {
+                                        request_consumed = user_data_migrator.intercept_broker_request(&broker_c, ws_tx_wrap.clone(), &mut request).await;
```

Comment:
What does the failure case  produce, and is it recoverable (maybe I'm asking questions that are not "now problems")

Reply:
If the `thunder_broker` has `user_data_migrator` installed, all requests will be intercepted for migration processing. 

`request_consumed` will be false for the following scenarios.

1) The Requested FireBolt API is not listed in the migration configuration file 
2) There are errors in the migration configuration entries.
3) Migration is not required, for example, if there are no entries in Legacy Storage.
4) The request is partially consumed. For example,  in the case of a setter Firebolt API, the migrator will set the new value in Legacy Storage  and allow the `thunder_broker` to continue the rest of the operation of updating the value in the new plugin.